clusterloader2/framework/framework.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/perf-tests/clusterloader2/api"
+	"k8s.io/perf-tests/clusterloader2/framework/util"
+)
+
+const (
+	automanagedNamespaceName = "namespace"
+	contentType              = "application/vnd.kubernetes.protobuf"
+	qps                      = 100
+	burst                    = 200
+)
+
+// Framework allows for interacting with Kubernetes cluster via
+// official Kubernetes client.
+type Framework struct {
+	automanagedNamespaceCount int
+	clientSet                 clientset.Interface
+	dynamicClient             dynamic.Interface
+}
+
+// NewFramework creates new framework based on given kubeconfig.
+func NewFramework(path string) (*Framework, error) {
+	config, err := util.LoadConfig(path)
+	if err != nil {
+		return nil, fmt.Errorf("config load failed: %v", err)
+	}
+	config.ContentType = contentType
```

Comment:
We should remove it from here. What you should do is:
- make LoadConfig function private one
- add one more function (sth like PrepareConfig(path) or sth like that) that will:
(a) call loadConfig
(b) set all additional stuff that we want.